### PR TITLE
feat: refactor usecore to new state context provider

### LINKED ISSUE #43: Refactor useCore to new state context provider
**Is your feature request related to a problem? Please describe.**
The package should adapt to the changes made in https://github.com/Tresjs/tres/issues/331

**Describe the solution you'd like**
We should refactor `useCore` and any use of the state.

### PR DESCRIPTION
**Pull request by Tinoooo**
Closes #43

### CODE REVIEW

File: docs/guide/effects/bloom.md [hunk 1]
```diff
@@ -22,13 +22,14 @@ import { EffectComposer, Bloom } from '@tresjs/post-processing'
...
+| -------------------- | ---------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `blendFunction`      | The blend function of this effect. This prop is not reactive.                                        |  [BlendFunction.SCREEN](https://github.com/pmndrs/postprocessing/blob/c3ce388be247916437a314f17748a75329d65df1/src/enums/BlendFunction.js#L40) |
+| `intensity`          | The intensity of the bloom effect.                                                                   | `1`                                                                                                                                            |
+| `blurPass`           | An efficient, incremental blur pass.                                                                 |                                                                                                                                                |
```
Comment (JaimeTorrealba) [inline docs/guide/effects/bloom.md:29]: These default values are undefined?

**Review by JaimeTorrealba — APPROVED** [review]

File: docs/guide/effects/bloom.md (hunk 1 above)
Comment (Tinoooo) [reply docs/guide/effects/bloom.md]: @JaimeTorrealba No, they were just missing. I removed all 3 props. `blurPass` doesn't even have a setter or constructor value. `width`and `height` are deprecated.

File: src/core/effects/Outline.vue [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
-import { useCore } from '../useCore'
-import { normalizeColor } from '@tresjs/core'
+import { normalizeColor, useTresContext } from '@tresjs/core'
 import { EffectPass, OutlineEffect } from 'postprocessing'
 import { effectComposerInjectionKey } from '../injectionKeys'
 import { inject, onUnmounted, shallowRef, watch, watchEffect, computed } from 'vue'
 
 import type { TresColor } from '@tresjs/core'
-import type { Object3D, Texture } from 'three'
+import type { Object3D, Scene, Texture } from 'three'
```
Comment (JaimeTorrealba) [inline src/core/effects/Outline.vue:8]: This Scene from three is for types?

**Review by JaimeTorrealba — APPROVED** [review]

**Review by alvarosabu — APPROVED** [review]

File: src/core/effects/Bloom.vue [hunk 3]
```diff
@@ -74,64 +52,50 @@ export type BloomProps = {
...
-    if (pass.value) {
-      disposePass()
-      createPass()
-      composer?.value?.addPass(toRaw(pass.value) as EffectPass)
-    }
-  },
-)
+  // blendFunction is not updated, because it has no setter in BloomEffect
+
+  effect.value.intensity = props.intensity !== undefined ? props.intensity : plainEffectPass.intensity
```
[the remark this answers is not in the document]
Comment (Tinoooo) [reply src/core/effects/Bloom.vue:85]: That's true. But that would make it even more lines 😬.

File: src/core/effects/Outline.vue (hunk 2 above)
Comment (Tinoooo) [reply src/core/effects/Outline.vue:8]: Yes, only the type is imported.

**Pull request merged**